docs/design/dp_network_policy_arktos_mizar_int.md

```diff
@@ -0,0 +1,77 @@
+# Enforcing Kubernetes Network Policy in an Arktos-Mizar Integrated Environment
+
+
+This document describes the design of enforcing Kubernetes network policy in Arktos-Mizar integrated environment, including the requirements, architecture, component changes and design considerations.
+
+## Requirements
+* Support standard [network policy object spec] (https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#networkpolicy-v1-networking-k8s-io) in Arktos
+
+## Architecture
+
+Below is the overall architecture of the network policy support in Arktos and Mizar: 
+
+![alt text](png/network_policy_arktos_mizar_int_architecture.png "Architecture")
+
+<p style="text-align: center;"> Figure 1 The overall architecture </p>
+
+## Workflow
+
+With our current design, network policy controller (in yellow) only watches network policy object events, and it is depended on ```mizar network policy operator (in red)``` to do the calculation and then update eBPF maps. For instance, when a network policy is created, ```network policy controller (in yellow)``` detects the event and lists network policy’s metadata, and then it composes network policy’s metadata into ```GRPC message``` and sends the message through ```GRPC client (on the arktos side)```. Then ```GRPC server (on the mizar side)``` receives the message and triggers network policy creation workflow which is operated by ```network policy operator (in red)```. Network policy operator then starts the calucation process described below. 
+
+### Network Policy Create/Update Calculation
+
+Update_and_retrieve_endpoint_names: 
+
+1. Update operator store for label_networkpolicies_store[label] = network_policy which is stored in mizar operator store. Note that operator store is a in-memory storage system in mizar.
+2.	Based on label vs network_policy dictionary in step 1, find all the pod’s names that satisfy label key-value set. NOTE: this step involves calling Kubernetes API server to get list of pods.

```

Comment:
Where in the code do we make this call today?

Reply:
https://github.com/CentaurusInfra/mizar/blob/dev-next/mizar/dp/mizar/workflows/builtins/networkpolicies/create.py#L43